app/forms/state_file/az_retirement_income_form.rb

```diff
@@ -0,0 +1,33 @@
+module StateFile
+  class AzRetirementIncomeForm < QuestionsForm
+    set_attributes_for :intake,
+                       :received_military_retirement_payment,
+                       :received_military_retirement_payment_amount,
+                       :primary_received_pension,
+                       :primary_received_pension_amount,
+                       :spouse_received_pension,
+                       :spouse_received_pension_amount
```

Comment:
I get that this is the standard practice atm but I don't get why we don't just use `delegate` instead;

```ruby
delegate :intake,
         :received_military_retirement_payment,
         :received_military_retirement_payment_amount,
         :primary_received_pension,
         :primary_received_pension_amount,
         :spouse_received_pension,
         :spouse_received_pension_amount
         to: :intake
```
 I guess we would need to delegate the writers as well as the readers but imo being a little closer to standard rails might be worth it. The code around `set_attributes_for` seems to be doing a lot of work, especially around separating and reconnecting attributes to particular models that I feel like we can shed with this approach. Not that I'm advising a refactor atm, just thinking about what we use `set_attributes_for` out loud.

Reply:
hm yeah could be a computer gardening topic! (you could do a spike and present your findings) `QuestionsForm` was lifted from another cfa project when we spun up the codebase and i haven't honestly given it much thought ever